### PR TITLE
[ADD] module board_eval_context

### PR DESCRIPTION
**Pull request by gurneyalex**
This module adds some useful keys in the evaluation context of board.board records, which can be useful when creating generic boards for all the users of an instance.

At the moment it allows using the following values in the domains of a board.board record:

* datetime: the datetime.datetime class
* date: the datetime.date class
* timedelta: the datetime.timedelta class
* timezone: the datetime.timezone class
* tzinfo: the datetime.tzinfo class
* relativedelta: the dateutil.relativedelta.relativedelta class
* uid: the ID of the current user

### CODE REVIEW

**Comment by gurneyalex** [comment]
Following suggestion by @legalsylvain  to propose the module here rather than in OCA/server-ux. 

Replaces https://github.com/OCA/server-ux/pull/591

**Review by legalsylvain — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: board_eval_context/models/board_board.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    def _get_eval_context(self):
+        """Prepare the context used when evaluating python code
+        :returns: dict -- evaluation context given to safe_eval
+        """
+        return {
+            "datetime": datetime.datetime,
+            "date": datetime.date,
+            "timedelta": datetime.timedelta,
+            "timezone": datetime.timezeon,
```
Comment (yankinmax) [inline board_eval_context/models/board_board.py]:
```suggestion
            "timezone": datetime.timezone,
```

**Comment by gurneyalex** [comment]
small error in the implementation: tools.safe_eval is the module not the function. Fixing

**Review by ivantodorovich — APPROVED** [review]

**Review by yankinmax — APPROVED** [review]

**Pull request merged**